### ISSUE TITLE
Convergence Issue of ShapeletModel

### ISSUE THREAD

**Issue by AStupidBear** [tslearn-team/tslearn#171]
**Describe the bug**
The accuracy of ShapeletModel is too low for multidimensional data.
```
Epoch 999/1000
300/300 [==============================] - 0s 25us/step - loss: 5.3731 - categorical_accuracy: 0.6667 - categorical_crossentropy: 5.3731
Epoch 1000/1000
300/300 [==============================] - 0s 25us/step - loss: 5.3731 - categorical_accuracy: 0.6667 - categorical_crossentropy: 5.3731
```
**To Reproduce**
```python
from tslearn.shapelets import ShapeletModel
from tslearn.generators import random_walk_blobs
x, y = random_walk_blobs(n_ts_per_blob = 100, sz = 38, d = 3, n_blobs = 3)
clf = ShapeletModel(max_iter = 1000, verbose = 1)
clf.fit(x, y)
```
**Expected behavior**
If `d` is set to `2`, the accuracy will be close to `1`

**Environment (please complete the following information):**
 - OS: Ubuntu 18.04
 - tslearn version 0.2.2

**Comment by rtavenar** [comment]
Hi @AStupidBear 

Thanks for your feedback.
Are you able to observe how this accuracy evolves as a function of the number of samples? 

```python
from tslearn.shapelets import ShapeletModel
from tslearn.generators import random_walk_blobs

for n in [100, 1000, 10000]:
    x, y = random_walk_blobs(n_ts_per_blob = n, sz = 38, d = 3, n_blobs = 3)
    clf = ShapeletModel(max_iter = 1000, verbose = 1)
    clf.fit(x, y)
```

Or whether this score is reproducible across seeds?

```python
from tslearn.shapelets import ShapeletModel
from tslearn.generators import random_walk_blobs

x, y = random_walk_blobs(n_ts_per_blob = 100, sz = 38, d = 3, n_blobs = 3)
for seed in range(5):
    clf = ShapeletModel(max_iter = 1000, verbose = 1, random_state=seed)
    clf.fit(x, y)
```

**Comment by AStupidBear** [comment]
The results are always bad.

**Comment by GillesVandewiele** [comment]
Hi,

Convergence issues can arise when the data is not normalized.

```python3
from tslearn.shapelets import ShapeletModel
from tslearn.generators import random_walk_blobs
from tslearn.preprocessing import TimeSeriesScalerMinMax
x, y = random_walk_blobs(n_ts_per_blob = 100, sz = 38, d = 3, n_blobs = 3)
x = TimeSeriesScalerMinMax().fit_transform(x)
clf = ShapeletModel(max_iter = 1000, verbose = 1)
clf.fit(x, y)
```

will produce:
```
Epoch 999/1000
300/300 [==============================] - 0s 122us/step - loss: 0.8076 - categorical_accuracy: 0.9933 - categorical_crossentropy: 0.8076
Epoch 1000/1000
300/300 [==============================] - 0s 125us/step - loss: 0.8072 - categorical_accuracy: 0.9933 - categorical_crossentropy: 0.8072
```

@rtavenar since the timeseries-wise normalization is idempotent (normalizing an already normalized timeseries will have no impact), we could perhaps always do this?

**Comment by rtavenar** [comment]
Hi @GillesVandewiele and thanks for the answer!

This is a general question. I'd prefer checking for standardization and raising alerts rather than enforcing them, since it could prove harmful in applications for which the standardization should not be the standard `TimeSeriesScalerMinMax`. But, once again, we should discuss that and come out with a general solution for such cases.

**Comment by johannfaouzi** [comment]
Or add a `standardize` parameter that controls the behavior. Default behavior could be True, since it seems to be important in order to find a good local minimum.

**Comment by GillesVandewiele** [comment]
I agree that this would be a nice addition. But I also agree with @rtavenar that we should make this configurable. I can imagine use cases where the magnitude of values of the shapelets really matter and that normalization would result in reduced performance.

On a sidenote: I have been pretty busy lately, but I will soon integrate the comments in my PR.

**Comment by johannfaouzi** [comment]
Adding a parameter would make it configurable, as the user will control the behavior! Default behavior could also be False to ensure backward compatibility.

**Comment by rtavenar** [comment]
Yep, adding a parameter would probably be the best option. And having a default to `False` (maybe with deprecation warning to change it later) would be reasonable.

**Comment by rtavenar** [comment]
This is similar in spirit to #237
